src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/CmsSignature.RSA.cs

```diff
@@ -108,60 +111,68 @@ private protected static bool SignCore(
                 RSASignaturePadding signaturePadding,
                 [NotNullWhen(true)] out byte[]? signatureValue)
             {
-                RSA certPublicKey = certificate.GetRSAPublicKey()!;
-
-                // If there's no private key, fall back to the public key for a "no private key" exception.
-                RSA? privateKey = key as RSA ??
-                    PkcsPal.Instance.GetPrivateKeyForSigning<RSA>(certificate, silent) ??
-                    certPublicKey;
-
-                if (privateKey is null)
+                using (GetSigningKey(key, certificate, silent, certificate.GetRSAPublicKey, out RSA? privateKey))
```

Comment:
Same here, `using` declaration would avoid all the indent diffs. This initialized at the top of the function and its lifetime should extend until the method returns.